fair/session.py

```diff
@@ -605,9 +605,9 @@ def close_session(self) -> None:
         with open(fdp_com.local_fdpconfig(self._session_loc), "w") as f:
             yaml.dump(self._local_config, f)
 
-        #TODO Kristian decide where this code should go
+        #TODO Kristian decide where this code should go and whether there is a better way of identifying _local_uri
         # Populate file type table
-        _local_uri = fdp_conf.get_local_uri()
+        _local_uri = self._global_config['registries']['local']['uri']
```

Comment:
Can I ask why have you changed this line? The function `get_local_uri` does this anyway, but also catches the exception to print it in the same style as all errors in the CLI (hides the traceback unless `--debug` is used) and ensures it can be called from anywhere. Using a global function is better than hardcoding as it means should the layout of things change, only one place has to be altered.

```python
def get_local_uri() -> str:
    _cfg = read_global_fdpconfig()
    try:
        return _cfg['registries']['local']['uri']
    except KeyError:
        raise fdp_exc.CLIConfigurationError(
            f"Expected key 'registries:local:uri' in local CLI configuration"
        )
```

Reply:
It doesn't do the same actually - I suspect because `get_local_uri()` relies on `read_global_fdpconfig()` which relies on the file being written rather than being in memory?

Reply:
@kzscisoft - I totally agree this is inelegant btw, happy to have a fix, but I just don't think it can involve `get_local_uri()`...

Reply:
> It doesn't do the same actually - I suspect because `get_local_uri()` relies on `read_global_fdpconfig()` which relies on the file being written rather than being in memory?

Hmm but is written a couple of lines above:

```python
with open(fdp_com.global_fdpconfig(), "w") as f:
            yaml.dump(self._global_config, f)
```

Is this write failing?

Reply:
It may seem a bit weird to re-read the same thing, but it is a failsafe in a way in that it tests the write actually did happen.

Reply:
Hah. Well, that's weird. It works now. I suspect that the file wasn't getting written out correctly because it was looking for a `description` key somewhere, and I got rid of that requirement - when I looked the file existed but was empty, so I assumed that it was part way through writing for some reason. Anyway, I'll happily undo this change then.